### PR TITLE
Kdoc fixer false-negative: not generating a Kdoc for several public functions

### LINKED ISSUE #965: Kdoc fixer false-negative: not generating a Kdoc for several public functions
diktat 1.0.0-RC

```
    fun hasNoChildren() = children.size == 0
    fun getFirstChild() = children.elementAtOrNull(0)
```

```
/**
 * @return
 */
fun hasNoChildren() = children.size == 0
// <------ where is a Kdoc?
fun getFirstChild() = children.elementAtOrNull(0)
```

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* fixed bug in MISSING_KDOC_ON_FUNCTION
* We are changing criterion for getters and setters. This is single line body function with element type CALL_EXPRESSION фтв REFERENCE_EXPRESSION or DOT_QUALIFIED_EXPRESSION with keyword _this_
Closes #965

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt [hunk 1]
```diff
@@ -251,12 +251,27 @@ class KdocMethods(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
-    private fun ASTNode.isSingleLineGetterOrSetter() = isGetterOrSetter() && (expressionBodyTypes.any { hasChildOfType(it) } || getBodyLines().size == 1)
+    private fun ASTNode.isSingleLineGetterOrSetter(): Boolean {
+        val isDotQualifiedExp = this.findChildByType(DOT_QUALIFIED_EXPRESSION)?.psi
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt:255]: Prefix `is` implies boolean type, but it's not

Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt:255]: You can cast it to a required type right away, e.g. using `?.let { it as KtDorQualifiedExpression }`

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt [hunk 2]
```diff
@@ -251,12 +251,27 @@ class KdocMethods(configRules: List<RulesConfig>) : DiktatRule(
...
+        val isThisExpression = isDotQualifiedExp != null && (isDotQualifiedExp as KtDotQualifiedExpression).receiverExpression.node.elementType == THIS_EXPRESSION
+        val isExpressionBodyTypes = expressionBodyTypes.any { hasChildOfType(it) }
+        return isGetterOrSetter() && (isExpressionBodyTypes || getBodyLines().size == 1 || isThisExpression)
+    }
 
     companion object {
         // expression body of function can have a lot of 'ElementType's, this list might be not full
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt:262]: This comment is no longer accurate

**Review by petertrr — CHANGES_REQUESTED** [review]
Please provide a more detailed description in the PR, to summarize what we are changing here. We are changing criterion for getters and setters and changing expression body detection, right?
Also please check out and fix newly discovered rules violations in diktat_snapshot workflow

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt (hunk 1 above)
Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt]: Fixed

**Review by petertrr — APPROVED** [review]

**Pull request merged**